### PR TITLE
Added possibility to configure and add toolbar buttons

### PR DESCRIPTION
**Pull request by bolelamx**
Example of configuring and using:

```
angular.module('myapp').config(['ngWigToolbarProvider', function(ngWigToolbarProvider) {
        ngWigToolbarProvider.setConfig({
            button: ['list1', 'list2', 'bold', 'italic', 'link'],
            format: ['p', 'h1', 'h2', 'h3'],
            other: ['source']
        });
}]);
```

Example of adding custom buttons:

```
angular.module('myapp').config(['ngWigToolbarProvider', function(ngWigToolbarProvider) {        
        ngWigToolbarProvider.addButton('underline',{title: 'Underline', command: 'underline', styleClass: 'nw-button--underline'});
	ngWigToolbarProvider.addButton('strikethrough');
}]);
```

### CODE REVIEW

File: src/javascript/app/ng-wig/ngWigDirective.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 angular.module('ngWig')
-  .directive('ngWig', function (ngWigToolbarProvider) {
```
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigDirective.js]: :+1:

**Comment by stevermeister** [comment]
could you add example to actual code, please?

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 2]
```diff
@@ -1,29 +1,96 @@
...
+                title: lowerCaseName[0].toUpperCase() + lowerCaseName.slice(1),
+                command: lowerCaseName,
+                styleClass: 'nw-button--' + lowerCaseName
+            };
+
+        itemLibrary[BUTTON_TYPE][name] = angular.extend({},defaultOptions, options);
+        addToConfig(BUTTON_TYPE, name);
+    };
+
+    this.addFormat = function (name, value) {
```
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: format dropdown would be plugin itself so no need to put it here

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 3]
```diff
@@ -1,29 +1,96 @@
...
+
+    var itemLibrary = {
+      button: {
+          list1: {title: 'Unordered List', command: 'insertunorderedlist', styleClass: 'nw-button--unordered-list'},
+          list2: {title: 'Ordered List', command: 'insertorderedlist', styleClass: 'nw-button--ordered-list'},
+          bold: {title: 'Bold', command: 'bold', styleClass: 'nw-button--bold'},
+          italic: {title: 'Italic', command: 'italic', styleClass: 'nw-button--italic'},
+          link: {title: 'Link', command: 'createlink', styleClass: 'nw-button--link'}
+      },
+      format: {
```
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: commented behind

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 4]
```diff
@@ -1,29 +1,96 @@
...
-        var toolbarButtons = [];
-        (list || defaultButtonsList).forEach(function(buttonKey) {
-          if(!buttonLibrary[buttonKey]) {
-            throw 'There is no "' + buttonKey + '" in your library. Possible variants: ' + Object.keys(buttonLibrary);
-          }
-          toolbarButtons.push(buttonLibrary[buttonKey]);
-        });
-        return toolbarButtons;
+angular.module('ngWig').provider('ngWigToolbar', function () {
+    var FORMAT_TYPE = 'format',
```
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: didn't get the idea of this variables

**Comment by bolelamx** [comment]
I removed format dropdow from toolbar provider and added examples of configuring and adding custom button to code. Please note that also need to add css styles for button icon.

**Pull request closed without merging**